Changing the template of an instance with volumes attached in terraform fails

It calls the update on the volumes and tries to detach the volume but the instance has been destroyed.
ex:
cloudca_volume.data_volume_43_2: Modifying...
  instance_id: "c6cfd6a4-ee15-4d86-b17d-7dade9c95ff1" => "de0518ef-81c6-427e-958f-64aa3170495a"
cloudca_volume.data_volume_43_4: Modifying...
  instance_id: "c6cfd6a4-ee15-4d86-b17d-7dade9c95ff1" => "de0518ef-81c6-427e-958f-64aa3170495a"
cloudca_volume.data_volume_43_3: Modifying...
  instance_id: "c6cfd6a4-ee15-4d86-b17d-7dade9c95ff1" => "de0518ef-81c6-427e-958f-64aa3170495a"
cloudca_volume.data_volume_43: Modifying...
  instance_id: "c6cfd6a4-ee15-4d86-b17d-7dade9c95ff1" => "de0518ef-81c6-427e-958f-64aa3170495a"



* cloudca_volume.data_volume_44: [ERROR] Received HTTP status code 400
[ERROR] Error Code: 2704, Message: This volume is not attached to existing instance, Context: {}

* cloudca_volume.data_volume_42: [ERROR] Received HTTP status code 400
[ERROR] Error Code: 2704, Message: This volume is not attached to existing instance, Context: {}

* cloudca_volume.data_volume_43_4: [ERROR] Received HTTP status code 400
[ERROR] Error Code: 2704, Message: This volume is not attached to existing instance, Context: {}

* cloudca_volume.data_volume_43: [ERROR] Received HTTP status code 400
[ERROR] Error Code: 2704, Message: This volume is not attached to existing instance, Context: {}

* cloudca_volume.data_volume_43_3: [ERROR] Received HTTP status code 400
[ERROR] Error Code: 2704, Message: This volume is not attached to existing instance, Context: {}

* cloudca_volume.data_volume_43_2: [ERROR] Received HTTP status code 400
[ERROR] Error Code: 2704, Message: This volume is not attached to existing instance, Context: {}

--- a/cloudca/resource_cloudca_volume.go
+++ b/cloudca/resource_cloudca_volume.go
@@ -145,10 +145,11 @@ func resourceCloudcaVolumeUpdate(d *schema.ResourceData, meta interface{}) error
 	d.Partial(true)
 	if d.HasChange("instance_id") {
 		oldInstanceId, newInstanceId := d.GetChange("instance_id")
+		curVolume, _ := ccaResources.Volumes.Get(d.Id())
 		volume := &cloudca.Volume{
 			Id: d.Id(),
 		}
-		if oldInstanceId != "" {
+		if oldInstanceId != "" && curVolume.InstanceId != "" {
 			err := ccaResources.Volumes.DetachFromInstance(volume)
 			if err != nil {
 				return err

